substrate/frame/message-queue/src/lib.rs

```diff
@@ -1554,12 +1564,16 @@ impl<T: Get<O>, O: Into<u32>> Get<u32> for IntoU32<T, O> {
 impl<T: Config> ServiceQueues for Pallet<T> {
 	type OverweightMessageAddress = (MessageOriginOf<T>, PageIndex, T::Size);
 
-	fn service_queues(weight_limit: Weight) -> Weight {
+	fn service_queues(weight_limit: Weight, context: ServiceQueuesContext) -> Weight {
 		let mut weight = WeightMeter::with_limit(weight_limit);
 
 		// Get the maximum weight that processing a single message may take:
 		let max_weight = Self::max_message_weight(weight_limit).unwrap_or_else(|| {
-			defensive!("Not enough weight to service a single message.");
+			// throw defensive message when service_queues is called from on_initialize
+			// don't throw message when service_queues is called from on_idle
```

Comment:
I'd add the reason here, that it doesn't matter if `on_idle` doesn't have enough weight